theano/gof/tests/test_graph.py

```diff
@@ -311,3 +322,87 @@ def test_eval(self):
                 "variable must have cache after eval")
         self.assertFalse(hasattr(pickle.loads(pickle.dumps(self.w)), '_fn_cache'),
                 "temporary functions must not be serialized")
+
+
+################
+# autoname     #
+################
+class TestAutoName:
+
+    def test_auto_name(self):
+        ## Re-init counter
+        Variable.__count__ = count(0)
```

Comment:
You should reset the old count after the test. Otherwise the test will change global value. Idem for all tests.

If fact, why change it? Why not read the current value and use that as the staring point instead of 0?